File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    while progresses:
        cnt = 0
        left_day = math.ceil((100 - progresses[0]) /speeds[0])
        for i in range(len(progresses)):
            progresses[i] += speeds[i] * left_day
        
        for i in range(len(progresses)):
            if progresses[0] >= 100:
                cnt += 1
                progresses.pop(0)
                speeds.pop(0)
            else:
                answer.append(cnt)
                break
        if len(progresses) == 0:
            answer.append(cnt)
                
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    const arr = []
    for(let i = 0; i < progresses.length; i++){
        let time = Math.ceil((100 - progresses[i]) / speeds[i])
        arr.push(time)
    }
    while(arr.length > 0){
        let cnt = 0
        let num = arr[0]
        while(arr.length > 0 && arr[0] <= num){
            arr.shift()
            cnt ++
        }
        answer.push(cnt)
        }
    return answer;
}